### PR TITLE
Dissolve Topology module and move files to nestkernel/spatial

### PR DESCRIPTION
**Pull request by jougs**

### CODE REVIEW

**Review by hakonsbm — APPROVED** [review]
:+1:

File: nestkernel/conn_parameter.cpp [hunk 1]
```diff
@@ -39,7 +39,7 @@
 nest::ConnParameter*
 nest::ConnParameter::create( const Token& t, const size_t nthreads )
 {
-  // Code grabbed from TopologyModule::create_parameter()
+  // Code grabbed from NestModule::create_parameter()
   // See there for a more general solution
```
Comment (stinebuu) [inline nestkernel/conn_parameter.cpp:43]: Are these two lines needed anymore?

Comment (jougs) [reply nestkernel/conn_parameter.cpp]: @stinebuu: You're right. Removed in 409b2803e.

**Review by stinebuu — APPROVED** [review]
# :partying_face:

**Pull request merged**